pkgs/applications/video/clapper/default.nix

```diff
@@ -0,0 +1,86 @@
+{ config, lib, stdenv, fetchFromGitHub
+,  glib
+,  gobject-introspection
+,  python3
+,  pkg-config
+,  ninja
+,  wayland
+,  wayland-protocols
+,  desktop-file-utils
+,  makeWrapper
+,  shared-mime-info
+,  wrapGAppsHook
+,  meson, gjs, gtk4, gst_all_1
+}:
+
+stdenv.mkDerivation rec {
+  pname = "clapper";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    owner  = "Rafostar";
+    repo   = pname;
+    rev    = version;
+    sha256 = "0gvn36njyhvz1hcywmdkla5as0qdqgc1bvkymmaipn26d80vxkdr";
+  };
+
+  nativeBuildInputs = [
+    desktop-file-utils # for update-desktop-database
+    glib
+    gobject-introspection
+    meson
+    ninja
+    makeWrapper
+    pkg-config
+    python3
+    shared-mime-info # for update-mime-database
+    wrapGAppsHook # for gsettings
+  ];
+
+  buildInputs = [
+    gjs
+    gst_all_1.gstreamer
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good
+    gst_all_1.gst-plugins-bad
+    gst_all_1.gst-plugins-ugly
+    gst_all_1.gst-libav
+    gst_all_1.gst-vaapi
+    gtk4
+    wayland
+    wayland-protocols
+  ];
+
+  postPatch = ''
+    patchShebangs build-aux/meson/postinstall.py
+  '';
+
+  mesonFlags = [
+    # TODO: https://github.com/NixOS/nixpkgs/issues/36468
+    "-Dc_args=-I${glib.dev}/include/gio-unix-2.0"
+  ];
+
+  postInstall = ''
+    mv $out/bin $out/nobin
```

Comment:
Why do we need this none FHS directory?

Reply:
Ideally we could just wrap the `com.github.rafostar.Clapper` binary, but this doesn't work because the original binary ends up with the name `.com.github.rafostar.Clapper-wrapped`. This is a problem because the binary reads `argv[0]` to determine which path it should use to find the package's, and thus looks for files in the wrong path: `../share/.com.github.rafostar.Clapper-wrapped/` which does not exist: https://paste.debian.net/1205618/

Thus we have to ensure the `com.github.rafostar.Clapper` retains its name, which we do by moving it to its own directory, `nobin/`.

Alternatively, I tried wrapping `com.github.rafostar.Clapper` to a binary named `clapper`, but this then breaks the desktop file: https://github.com/Rafostar/clapper/blob/eb60d036db16719f60be45983f236c5eb878cbe0/data/com.github.rafostar.Clapper.desktop#L7

Reply:
Either update the desktop file or the argv0 usage.

Reply:
@SuperSandro2000 I tried to set `argv0` in the wrapper instead.

<details>
  <summary>Diff</summary>

```diff
diff --git a/pkgs/applications/video/clapper/default.nix b/pkgs/applications/video/clapper/default.nix
index eda4712ff55..78449b71f60 100644
--- a/pkgs/applications/video/clapper/default.nix
+++ b/pkgs/applications/video/clapper/default.nix
@@ -17,6 +17,7 @@
 , gjs
 , gtk4
 , gst_all_1
+, libsoup
 }:
 
 stdenv.mkDerivation rec {
@@ -55,6 +56,7 @@ stdenv.mkDerivation rec {
     gtk4
     wayland
     wayland-protocols
+    libsoup
   ];
 
   postPatch = ''
@@ -66,12 +68,15 @@ stdenv.mkDerivation rec {
     "-Dc_args=-I${glib.dev}/include/gio-unix-2.0"
   ];
 
-  postInstall = ''
-    mv $out/bin $out/nobin
+  dontWrapGApps = true;
+
+  postFixup = ''
     # Clapper reads argv[0] and expects its to be com.github.rafostar.Clapper.
-    # Thus, we cannot use wrapProgram: doing so would make argv[0] == .com.github.rafostar.Clapper-wrapped.
-    makeWrapper $out/nobin/com.github.rafostar.Clapper $out/bin/com.github.rafostar.Clapper \
-      --prefix GST_PLUGIN_SYSTEM_PATH_1_0 : "$GST_PLUGIN_SYSTEM_PATH_1_0"
+    for bin in $out/bin/*; do
+      wrapGApp $bin \
+        --prefix GST_PLUGIN_SYSTEM_PATH_1_0 : "$GST_PLUGIN_SYSTEM_PATH_1_0" \
+        --argv0 $(basename $bin)
+    done
   '';
 
   meta = with lib; {
```

</details>

This doesn't seem to work:

```
(.com.github.rafostar.Clapper-wrapped:61082): Gtk-WARNING **: 23:40:18.528: Theme parsing error: <broken file>:1:1: Error opening file /nix/store/vyk63yz3s5vb08fr7bnalynksing7nc6-clapper-0.3.0/share/.com.github.rafostar.Clapper-wrapped/css/styles.css: No such file or directory

(.com.github.rafostar.Clapper-wrapped:61082): Gtk-ERROR **: 23:40:18.530: failed to add UI from file /nix/store/vyk63yz3s5vb08fr7bnalynksing7nc6-clapper-0.3.0/share/.com.github.rafostar.Clapper-wrapped/ui/clapper.ui: Failed to open file ?/nix/store/vyk63yz3s5vb08fr7bnalynksing7nc6-clapper-0.3.0/share/.com.github.rafostar.Clapper-wrapped/ui/clapper.ui?: No such file or directory
Trace/breakpoint trap (core dumped)
```

Reply:
Fixed in latest PR, which uses 0.4, which doesn't need these hacks.